### PR TITLE
docs: let's discuss threats

### PR DESCRIPTION
**Pull request by varovainen**

### CODE REVIEW

**Comment by Zhenya-Barannik** [comment]
For  kampela_questions.md :

## Decision 9. Software updates possibilities
Should Kampela be designed with software updates in mind? Even with prior testing, there is a chance that significant vulnerabilities or bugs will be found following the Kampela release.

Pros:
- it is possible to fix vulnerabilities and bugs after the release.

Cons:
- software updates introduce a new attack vector.
- we need a separate external program to handle the software updates.

## Decision 10. Port for charging, debugging, and keyboard connection 
Kampela must be supplied by energy in one way or another. NFC charging is an option, but the device used for routine NFC communications (e.g. mobile phone) in general will not be able to charge Kampela. It will also be impractical to buy an NFC charging device just to charge Kampela.

A physical port (e.g USB type-C) may be very beneficial for Kampela. Apart from charging, it can be used for fast debugging while in development and also for connecting the keyboard. If entering a 20-word phrase using two buttons is so cumbersome, imagine what it would take to enter a base 58 number or an ASCII string for a cyrillic derivation path; this is nothing less than a disaster!

Pros:
- easy to charge
- easy to get data from and into the device, especially useful while in development.

Cons:
- introduces a new attack vector. The user must make sure to not use a malicious cable.
- more sophisticated hardware and software

Most security issues will be solved by decoupling the physical port from the main data stream in the shipped version of Kampela. In this case, port will be usable only for charging and input (as redundant a system that substitutes the buttons), but not for sending any commands.

**Comment by Zhenya-Barannik** [comment]
for kampela_security_formalized.md :

# 7 Kampela is compromised on the way to the end user (hardware and/or software)
How end user can verify the absence of tampering when buying a device from a third party? For example, it is possible to imitate holographic stickers and reverse engineer physical procedures. Such cases are possible even at the "prohibitively high" cost of such modifications, compared to the cost of the individual Kampela device. Imitations may not necessarily need to look perfect, as the end user may not know what to look for.

In attack case # 1 Kampela first receives data and then falls into the hands of a malicious party. In this case, Kampela first falls into the hand's malicious party and then gets to the end user. A way to check the absence of tampering will protect against this kind of attack.

Base effort: High

Motivation: Get user to enter data on a compromised device, thus exposing data and assets.

## Integrity

7-I1 (also see 5-I1)
Attacker action: Modify Kampela hardware to transfer private data with NFC/radio.
Effort: High.
Incentive: High.

7-I2 (also see 3-I2)
Attacker action: Modify Kampela hardware to display false transaction details when certain conditions are met.
Effort: High.
Incentive: High.

7-I3 (also see 2-A1)
Attacker action: Modify Kampela to break/corrupt data after some conditions are met. In this case, a ransom may be requested.
Effort: High.
Incentive: Low.

7-I4 (also see 5-I4)
Attacker action: Modify Kampela to disable protection, get the user to use it, and then steal Kampela. Seems possible only if the identity of the user is known to begin with.
Effort: High.
Incentive: High.
## Availability
There is a motivation to maintain full availability so that the user is not suspicious (at least in some cases).

# 8 Kampela is compromised during the software update
Motivation: Compromise the device that is in current use, thus exposing data and assets.

## Integrity

(see attacks 2-I2 to 2-I6, they can also be realized in this case)

8-I1 
Attacker action: Modify software updater program (companion?) to compromise internal Kampela software. 
Effort: High.
Incentive: High.

This includes, but is not limited to
* modification enabling Kampela to transfer private data to the compromised software updater program;
* modification enabling Kampela to transfer private data with NFC;
* modification enabling Kampela to accept any companion device;
* modification enabling Kampela to display false information;
* modification to corrupt data and/or break Kampela, a ransom may also be requested in this case.

## Availability
There is a motivation to maintain full availability so that the user is not suspicious (at least in some cases).

File: docs/development/security/kampela_security_formalized.md [hunk 1]
```diff
@@ -292,23 +292,27 @@ In case of companion app (or actually any wallet's) breach, confidentiality dama
```
Comment (yury171) [inline docs/development/security/kampela_security_formalized.md:496]: I think that attacks need to be considered additionally in the context of the owner - the device
1 Attacks or impacts hidden or unknown to the Owner
this is what follows in the text
2 Attacks or impacts using the Owner himself
These are any KNOWN TO THE OWNER AND REGISTERED IN THE SYSTEM facts of the relationship of unauthorized persons with the device (here, loss, direct transfer, etc.).
here i see 2 options
a) Enabling a direct disposal protocol (lost - returned, service life expired - no guarantee of maintaining functionality, etc.) These are most likely options for administrative conflict resolution and also our maintenance headache, but if the issue under item b is resolved - then all the same the protocol of remote utilization.
b) Enabling a remote disposal protocol (the technical possibilities of self-destruction of either the device or its contents should be considered, excluding the very possibility of using the contents of the device).

File: docs/development/security/kampela_security_formalized.md [hunk 2]
```diff
@@ -292,23 +298,27 @@ In case of companion app (or actually any wallet's) breach, confidentiality dama
```
Comment (yury171) [inline docs/development/security/kampela_security_formalized.md:496]: I think that attacks need to be considered additionally in the context of the owner - the device
1 Attacks or impacts hidden or unknown to the Owner
this is what follows in the text
2 Attacks or impacts using the Owner himself
These are any KNOWN TO THE OWNER AND REGISTERED IN THE SYSTEM facts of the relationship of unauthorized persons with the device (here, loss, direct transfer, etc.).
here i see 2 options
a) Enabling a direct disposal protocol (lost - returned, service life expired - no guarantee of maintaining functionality, etc.) These are most likely options for administrative conflict resolution and also our maintenance headache, but if the issue under item b is resolved - then all the same the protocol of remote utilization.
b) Enabling a remote disposal protocol (the technical possibilities of self-destruction of either the device or its contents should be considered, excluding the very possibility of using the contents of the device).

**Review by yury171 — COMMENTED** [review]
I think that attacks need to be considered additionally in the context of the owner - the device
1 Attacks or impacts hidden or unknown to the Owner
this is what follows in the text
2 Attacks or impacts using the Owner himself
These are any KNOWN TO THE OWNER AND REGISTERED IN THE SYSTEM facts of the relationship of unauthorized persons with the device (here, loss, direct transfer, etc.).
here i see 2 options
a) Enabling a direct disposal protocol (lost - returned, service life expired - no guarantee of maintaining functionality, etc.) These are most likely options for administrative conflict resolution and also our maintenance headache, but if the issue under item b is resolved - then all the same the protocol of remote utilization.
b) Enabling a remote disposal protocol (the technical possibilities of self-destruction of either the device or its contents should be considered, excluding the very possibility of using the contents of the device).

**Comment by yury171** [comment]
I think that attacks need to be considered additionally in the context of the owner - the device
1 Attacks or impacts hidden or unknown to the Owner
this is what follows in the text
2 Attacks or impacts using the Owner himself
These are any KNOWN TO THE OWNER AND REGISTERED IN THE SYSTEM facts of the relationship of unauthorized persons with the device (here, loss, direct transfer, etc.).
here i see 2 options
a) Enabling a direct disposal protocol (lost - returned, service life expired - no guarantee of maintaining functionality, etc.) These are most likely options for administrative conflict resolution and also our maintenance headache, but if the issue under item b is resolved - then all the same the protocol of remote utilization.
b) Enabling a remote disposal protocol (the technical possibilities of self-destruction of either the device or its contents should be considered, excluding the very possibility of using the contents of the device).

**Review by Zhenya-Barannik — APPROVED** [review]
All is good. Also, I finally figured out how to push a local edit to the pull-request branch.

**Pull request merged**